### ISSUE TITLE
Bringing languages to a single form[Feature]: 

### ISSUE THREAD

**Issue by KOLIWAN** [fast4x/RiMusic#2008]
### Description

RiMusic is a multilingual app and I want to suggest an idea to improve the user experience for users from all countries.
I noticed that there are a lot of options for translating language names. In the screenshot below you can see that the app is now in Russian, in the list of available languages some of the languages:
1) Translated into English (bad).
2) Translated into English but in brackets the language self-name is written in the same language (better).
3) Translated to the selected language (in my case Russian) and in brackets the self-name of the language written in the same language (excellent!).

![Screenshot_20240525_133426](https://github.com/fast4x/RiMusic/assets/76595855/28e13247-a1d9-4c1e-960e-33c47391f723)

### Suggested Solution

I suggest the following solution: all self-names of languages are separately specified in the source code. When selecting the application language, they will be listed as **<Language_name_translated> (Self-name of language)** (as the variants marked in green on the screenshot).

This solves a lot of problems: if the user accidentally selects the wrong language, he can always find the right one in the list, and now it can be hard, because many language names are translated differently. It will also improve the user experience, the language selection menu will be neater. Also translators will only need to translate the name of the language and not have to google the correct spelling of the self-name. 

Below I have prepared a list of all available languages in the app and their self-names. If I have translated something wrong, write in the comments, I will correct it.

Afrikaans - Afrikaans
Arabic - اَلْعَرَبِيَّةُ
Bashkir - Башҡорт
Catalan - Català
Danish - Dansk
English - English
Esperanto - Esperanto
Estonian - Eesti keel
Chinese simplified - 汉语
Chinese traditional - 漢語
Czech - Čeština
Dutch - Nederlands
Finnish - Suomi
French - Français
German - Deutsch
Greek - Ελληνικά
Hebrew - עִבְרִית
Hindi - हिन्दी
Hungarian - Magyar
Italian - Italiano
Indonesian - Bahasa Indonesia
Japaneese - 日本語
Korean - 한국어
Odia - ଓଡ଼ିଆ
Persian - فارْسِى
Polish - Polski
Portuguese, Brazilian - Português brasileiro
Portuguese - Português
Romanian - Română
Russian - Русский
Serbian Cyrillic - Вуковица
Serbian Latin - Srpski
Sinhala - සිංහල
Spanish - Español
Swedish - Svenska
Telugu - తెలుగు
Turkish - Türkçe
Ukrainian - Українська
Vietnamese - Tiếng Việt

It would also be nice if in the strings.xml file the developer would label each language with the lang_LANGUAGENAME key. Now it can be just the name of the language 

![image](https://github.com/fast4x/RiMusic/assets/76595855/9bdda490-b7e7-4b60-8b07-2dada7e7ad6c)

or lang_language

![image](https://github.com/fast4x/RiMusic/assets/76595855/7a4e8c8a-2a5b-45a8-b49c-919e667c0725)

It would also be nice if they were all next to each other, now they are scattered all over the file.

**Comment by ikanakova** [comment]
fast4x wanted, when we started with translations, that each translator should decide this himself, so I have it in Czech - the name of the language in Czech (the name of the language in that language), e.g. Ruština (Русский). I translated some languages in English in Crowdin, so it's there too.

**Comment by dAtA-TRoN** [comment]
Nice addition. Love those little details.

**Comment by ikanakova** [comment]
My suggestion:
to make entities (see: https://stackoverflow.com/questions/4746058/reference-one-string-from-another-string-in-strings-xml) where each language would be in the original, e.g.:
```
<!ENTITY russian "Русский">
<!ENTITY english "English">
<!ENTITY czech "Čeština">
```
And then in each language it would be:
```
    <string name="en_Russian">Russian (&russian;)</string>
```

```
    <string name="cs_Russian">Ruština (&russian;)</string>
```

As long as it's compatible with Crowdin...

**Comment by KOLIWAN** [comment]
> My suggestion: to make entities (see: https://stackoverflow.com/questions/4746058/reference-one-string-from-another-string-in-strings-xml) where each language would be in the original, e.g.:
> 
> ```
> <!ENTITY russian "Русский">
> <!ENTITY english "English">
> <!ENTITY czech "Čeština">
> ```
> 
> And then in each language it would be:
> 
> ```
>     <string name="en_Russian">Russian (&russian;)</string>
> ```
> 
> ```
>     <string name="cs_Russian">Ruština (&russian;)</string>
> ```
> 
> As long as it's compatible with Crowdin...

Yep and also I think it would be super clean and nice if when selecting a language, its self-name was not duplicated 2 times like 
Русский (Русский)
or 
Čeština (Čeština)
or
English (English)
so in each xml file of a language the line with its name will look like this
`<string name="ru_Russian">Русский</string>`
or
`<string name="cs_Czech">Čeština</string>`
or
`<string name="en_English">English</string>`
without (&<language_name>;) part
